### PR TITLE
Tidy PO lexer

### PR DESCRIPTION
**Pull request by VladislavsPerkanuks**
Main reason of this PR

```
// Given example
msgid ""
"line1"
"line"

// Previous behaviour
token == "line1 line2"

// New behaviour
token == "\nline1\nline2"

// or raw (for more visuals)
`
line1
line2
`
```

Also fixed some strange behavior when newline character would be interpreted as an actual newline, not a text
e.g.
```
msgid ""
"line1\n"
"line2\n"

// Previous behaviour
token == "line1 \nline2\n" // not sure but something like that, I hope idea is clear

// New behaviour
token == "\nline1\\n\nline2\\n"

// or raw (for more visuals)
`
line1\n
line2\n
`


```

Also fixed an issue, in a situation, when the header is not defined in TokenType, then the parser tries to parse it as multiline msgid, msgstr e.t.c, and breaks. Either way it should throw an error, but in current implementation it thorws incorrect one

Also some tidy to lexer, to decrease technical debt :)

superset.pot | model.Translation |
---|---|
image |  image

### CODE REVIEW

File: pkg/convert/pot.go [hunk 1]
```diff
@@ -344,7 +344,7 @@ func convertPluralsToMessageString(plurals []string) string {
 
 	for i, plural := range plurals {
 		plural = escapeSpecialChars(plural)
-		line := strings.ReplaceAll(strings.TrimSpace(plural), "%d", "{$count}")
```
Comment (VladislavsPerkanuks) [inline pkg/convert/pot.go:347]: We cannot lose information

File: pkg/pot/parse.go [hunk 2]
```diff
@@ -66,9 +66,6 @@ func TokensToPo(tokens []Token) (Po, error) {
 			}
 		}
 
-		replacer := strings.NewReplacer("\\n ", "\n", "\\n", "\n")
-		token.Value = replacer.Replace(token.Value)
```
Comment (VladislavsPerkanuks) [inline pkg/pot/parse.go:70]: no clue what that was for :)

File: pkg/convert/pot_test.go [hunk 3]
```diff
@@ -1005,7 +1005,7 @@ func TestFromPot(t *testing.T) {
 				Messages: []model.Message{
 					{
 						ID:          "Hello",
-						Message:     "{Hello, world!\nvery long string\n}",
+						Message:     "{\nHello, world!\\\\n\nvery long string\\\\n}",
```
Comment (janishorsts) [inline pkg/convert/pot_test.go:1008]: Why so many escapes?

Escaped `\n` is escaped as string literal `\\n` in golang.
Isn't `\n` escaped as`\\n` in n MF2, too?


```suggestion
						Message:     "{\nHello, world!\\n\nvery long string\\n}",
```

Comment (janishorsts) [reply pkg/convert/pot_test.go:1008]: Ok, after some thinking - `\n` should be escaped in `\\n` in MF2, right? Now, to translate it into golang string literal we have to escape both slashes - `\\\\n`.

Comment (VladislavsPerkanuks) [reply pkg/convert/pot_test.go:1008]: po has raw text:
`"Hello world\n"` (\n is not newline here, but \\\n)
after converting it to golang string it becomes
`"Hello world\\n"`
as message format tells us that in the text backslashes must be escaped it becomes:
`"Hello world\\\\n"`

Comment (VladislavsPerkanuks) [reply pkg/convert/pot_test.go:1008]: in one image
image

File: pkg/pot/lex.go [hunk 4]
```diff
@@ -215,169 +99,140 @@ func Lex(r io.Reader) ([]Token, error) {
...
 
-// hasValidPrefix determines whether the provided line string has a valid prefix
-// by iterating through a collection of valid prefixes.
-func hasValidPrefix(line string) bool {
-	for _, prefix := range validPrefixes {
-		if strings.HasPrefix(line, prefix) {
-			return true
-		}
-	}
+// trimQuotes removes first and last double quotes from the string, if they exist.
```
Comment (janishorsts) [inline pkg/pot/lex.go:232]: No need to create a helper func. Simply use directly `strings.Trim(v, `"`).

PS. There is a side effect that I don't like - quotes are not optional when parsing. They are required.

Comment (janishorsts) [reply pkg/pot/lex.go:232]: I mean, we should error if there are no quotes present.

Comment (VladislavsPerkanuks) [reply pkg/pot/lex.go:232]: they are required in msgid, msgstr, and in headers
but for example in comments they are not.

Also they have differences

as in 


```
"Language: en-US\n" // prefix: Language: Value: en-US\n"
msgid "some text"   // prefix msgid: Value: "some-text"
#, python-format    // prefix:  # Value: python-format
```

Comment (janishorsts) [reply pkg/pot/lex.go:232]: We have to handle it correctly.

Comment (VladislavsPerkanuks) [reply pkg/pot/lex.go]: reverted to simple strings.Trim()

reason: Too many scenarios to handle.  Also case where file is formatted correctly is more common, than incorrectly (Pareto). So I just go with that, to save time

File: pkg/pot/lex.go [hunk 5]
```diff
@@ -197,9 +82,9 @@ func Lex(r io.Reader) ([]Token, error) {
 			continue
 		}
 
-		token, err := parseLine(line, &tokens)
+		token, err := parseLine(line, tokens)
 		if err != nil {
-			return nil, fmt.Errorf("parse line %d: %w", lineNumber, err)
+			return nil, fmt.Errorf("parse line \"%s\": %w", line, err)
```
Comment (janishorsts) [inline pkg/pot/lex.go:87]: alternatively, easier to read with backquotes

```suggestion
			return nil, fmt.Errorf(`parse line "%s": %w`, line, err)
```

**Review by janishorsts — APPROVED** [review]

**Pull request merged**